### PR TITLE
Feature/use html5 progress element

### PR DESCRIPTION
**Pull request by Samueljoli**
## Description

Change the underlying element used in `ProgressBar` from `div` to `progress`.

**NOTE:**
This PR is being opened prematurely. Mainly because I would like to open a discussion around my findings.

Some things to consider with using `<progress>`
1. Each browser provides it's own set of pseudo classes to style the status meter within the progress bar. So styling can only be applied via css. Inline styling will not work, therefore providing your own color is disabled, hence why that test has been skipped.
2. In our `scss` file we would have to account for each vendor prefix.
3. Support is ok across browsers and versions.

If there is any suggestions around being able to color the status meter I would greatly appreciate it 😄 

screen shot 2017-08-10 at 11 17 37 am

## Motivation and Context

Consistent semantic markup
## Types of changes

- [ ] Documentation
- [ ] Bug fix (non-breaking change which fixes an issue. you didn't modify existing tests)
- [ ] New feature (non-breaking change which adds functionality. you added at least one new test)
- [ ] Breaking change (fix or feature that would cause existing functionality to change. you had to modify existing tests.)
- [ ] Does this introduce a new pattern/package?

## Checklist:

- [ ] I have read the [**CONTRIBUTING**](https://github.com/xogroup/union/blob/master/.github/CONTRIBUTING.md) document.
- [ ] I have updated the documentation as needed.
- [ ] I have added tests to cover my changes.

## Reviewer Checklist

- [ ] Is change documented in a release draft? (Only if there are plans to merge PR)

### CODE REVIEW

**Comment by Samueljoli** [comment]
@GeorgeTaveras1231 Hey, here's my findings. Maybe you have some suggestions around getting custom meter color functionality into this PR. Or maybe the effort isn't worth it.

**Comment by GeorgeTaveras1231** [comment]
@Samueljoli we tried this in our end and it seems to work, but changing the `-webkit-progress-value`'s background color to `currentColor` allows consumers to overide the color via inline styles.

```scss
progress::-webkit-progress-value {
  background-color: currentColor;
}
```

Then the following works:

```html
<progress style="color: green;" value="0.5"></progress>
```

**Comment by GeorgeTaveras1231** [comment]
Also, support for `currentColor` is pretty good: [Can I Use](http://caniuse.com/#search=currentColor)

File: packages/@xo-union/tk-component-progress-bar/src/components/ProgressBar/index.jsx [hunk 1]
```diff
@@ -8,12 +8,21 @@ const ProgressBar = ({ progress, color }) => {
...
+      color="green"
+      className={styles.progress}
+      max="100"
+      value={progressWidth}
+    >
+
+      <div className={styles['progress-bar']} >
+        <div
+          className={styles['progress-fallback']}
+          style={{ width: `${progressWidth}%`, backgroundColor: color || null }}
```
Comment (hsbacot) [inline packages/@xo-union/tk-component-progress-bar/src/components/ProgressBar/index.jsx]: use default props to default color to null

File: packages/@xo-union/tk-component-progress-bar/src/components/ProgressBar/index.jsx [hunk 2]
```diff
@@ -8,12 +8,21 @@ const ProgressBar = ({ progress, color }) => {
...
 
   return (
-    <div className={styles['progress-bar']} >
-      <div
-        className={styles.progress}
-        style={{ width: `${progressWidth}%`, backgroundColor: color || null }}
-      />
-    </div>
+    <progress
+      color="green"
```
Comment (hsbacot) [inline packages/@xo-union/tk-component-progress-bar/src/components/ProgressBar/index.jsx]: Should this be hardcoded to green?

File: packages/@xo-union/tk-component-progress-bar/src/components/ProgressBar/browser-spec.jsx [hunk 3]
```diff
@@ -11,7 +11,7 @@ describe('<ProgressBar />', () => {
     return getProgressBar(props).find(progressSelector).node;
   };
 
-  it('should set the progress bar color', () => {
+  xit('should set the progress bar color', () => {
```
Comment (hsbacot) [inline packages/@xo-union/tk-component-progress-bar/src/components/ProgressBar/browser-spec.jsx]: How come this test is pending?

**Comment by Samueljoli** [comment]
@hsbacot @GeorgeTaveras1231 This PR is good to Review. I guess the only outstanding thing, is if your ok with accounting for all the vendor prefixes.

**Comment by GeorgeTaveras1231** [comment]
@Samueljoli I am ok with that in the mean time, we opened #250 to simplify this in the future.

**Pull request merged**